_posts/2025-05-21-secure-mcp-client.adoc

```diff
@@ -0,0 +1,486 @@
+
+For example, let's temporarily update the diagram by removing the `AI Gemini`, replacing `MCP Client` with `REST Client`, `MCP Server` with `Poem Creator service` and `GitHub` with `OAuth2`:
+
+image::typical_oauth2_authorization.png[Typical OAuth2 Authorization,align="center"]
+
+You will very likely find similarities between this diagram and what you do in your projects. It is the OAuth2 authorization code flow is action: the user logs in to the application and authorizes it to access another service offering a poem creation on the user's behalf.
+
+The demo shows that Quarkus MCP Client can work effectively in such architectures by being able to use access tokens acquired during the user login, without you having to write any custom code.
+
+We are now ready to start working on the `Secure MCP Client Server` demo.
+
+You can find the complete project source in the https://github.com/quarkiverse/quarkus-langchain4j/tree/main/samples/secure-mcp-sse-client-server[Quarkus LangChain4j Secure MCP Client Server sample].
+
+[[create-mcp-server]]
+== Step 1 - Create and start MCP server
+
+First, let's create a secure Quarkus MCP SSE server. 
+
+[NOTE]
+====
+If you already created the MCP server https://quarkus.io/blog/secure-mcp-sse-server/#initial-mcp-server[as described] in the the https://quarkus.io/blog/secure-mcp-sse-server/[Getting ready for secure MCP with Quarkus MCP Server] blog post, then you will find instructions below familiar and should be able to reuse the project you created earlier with minor updates.  
+====
+
+MCP server requires authentication to establish Server-Sent Events (SSE) connection and also when invoking the tools. Additionally, the MCP server endpoint that provides access to tools requires that the security identity has a `read:name` permission.
+
+[[mcp-server-dependencies]]
+=== MCP server maven dependencies
+
+Add the following dependencies:
+
+[source,xml]
+----
+<dependency>
+    <groupId>io.quarkiverse.mcp</groupId>
+    <artifactId>quarkus-mcp-server-sse</artifactId> <1>
+    <version>1.1.1</version>
+</dependency>
+<dependency>
+    <groupId>io.quarkus</groupId>
+    <artifactId>quarkus-oidc</artifactId> <2>
+</dependency>
+<dependency>
+    <groupId>io.quarkus</groupId>
+    <artifactId>quarkus-hibernate-reactive-panache</artifactId> <3>
+</dependency>
+<dependency>
+    <groupId>io.quarkus</groupId>
+    <artifactId>quarkus-reactive-pg-client</artifactId> <3>
+</dependency>
+----
+<1> `quarkus-mcp-server-sse` is required to support MCP SSE transport.
+<2> `quarkus-oidc` is required to secure access to MCP SSE endpoints. Its version is defined in the Quarkus BOM.
+<3> `quarkus-hibernate-reactive-panache` and `quarkus-reactive-pg-client` are required to support the <<security-identity-augmentation>>. Their versions are defined in the Quarkus BOM.
```

Comment:
do we really need to use hibernate-reactive for this?

Reply:
@maxandersen It was the existing demo that we did awhile back with Stuart and Steph that I copied some code from, and the augmentor code is very compact as a result.

But I can try and see if it now can be done ina  compact way without the reactive variation